test-data/unit/check-unreachable-code.test

```diff
@@ -631,11 +631,12 @@ class Child(Parent):
     def bar(self) -> int:
         if 1:
             self = super(Child, self).something()
-            reveal_type(self)       # N: Revealed type is "__main__.Child"
+            reveal_type(self)       # N: Revealed type is "Any"
+            # TODO: we should probably make this unreachable similar to above.
```

Comment:
I think this is https://github.com/python/mypy/issues/18386

Reply:
It is a bit more tricky than that. In that example:
```python
x: int
if is None: ...
```
the branch is obviously unreachable. But if I say:
```python
x: Any
if x is None: ...
```
then technically it is reachable because `x: Any = None` is valid, but (highly) unlikely.